daml-lf/archive/src/stable/protobuf/com/daml/daml_lf_1_15/daml_lf_1.proto

```diff
@@ -0,0 +1,1808 @@
+
+  // An interface instance defined in this template.
+  // *Available in versions >= 1.15*
+  message Implements {
+    TypeConName interface = 1;
+    InterfaceInstanceBody body = 2;
+  }
+
+  // The type constructor for the template, acting as both
+  // the name of the template and the type of the template argument.
+  oneof tycon {
+    DottedName tycon_dname = 1;
+
+    // *Must be a valid interned name*
+    int32 tycon_interned_dname = 12; // *Available in versions >= 1.7*
+  }
+
+  // Name to which the template argument is bound.
+  oneof param {
+    // *Must be a valid identifier*
+    string param_str = 2; // *Available in version < 1.7*
+
+    // *Must be a valid interned identifier*
+    int32 param_interned_str = 11; // *Available in versions >= 1.7*
+  }
+
+  // NOTE(MH): The new runtime authorization check for Daml 1.0 does not rely
+  // on the stakeholder signatures produced by the obligables computation
+  // anymore but uses the interpreter to compute the signatories and
+  // stakeholders of contract instances.
+  // REMOVED: TemplateStakeholders stakeholders = 3;
+  reserved 3;
+
+  // Pre-condition that the template argument must satisfy.
+  // When present, it has type `Bool` and the template parameter in scope.
+  // *Optional*, interpreted as 'True' if undefined
+  Expr precond = 4;
+
+  // The signatories of the contract. They have type `List Party` and the
+  // template parameter in scope.
+  Expr signatories = 5;
+
+  // The agreement text associated with the contract. It has type `Text` and
+  // the template parameter in scope.
+  Expr agreement = 6;
+
+  // The choices available in the resulting contract.
+  repeated TemplateChoice choices = 7;
+
+  // The observers of the contract. They have type `List Party` and the
+  // template parameter in scope.
+  Expr observers = 8;
+
+  Location location = 9;
+
+  // The key definition for the template, if present
+  DefKey key = 10; // optional
+
+  // Interface instances defined in this template.
+  repeated Implements implements = 13; // *Available in versions >= 1.15*
+}
+
+// Interface method definition.
+// *Available in versions >= 1.15*
+message InterfaceMethod {
+  Location location = 1;
+  int32 method_interned_name = 2;
+  Type type = 3;
+}
+
+// Interface definition.
+// *Available in versions >= 1.15*
+message DefInterface {
+
+  // An interface instance defined in this interface.
+  message CoImplements {
+    TypeConName template = 1;
+    InterfaceInstanceBody body = 2;
+  }
+
+  Location location = 1;
+  int32 tycon_interned_dname = 2;
+  repeated InterfaceMethod methods = 3;
+
+  // Binder for interface value ("this") in precond and fixed choices.
+  int32 param_interned_str = 4;
+
+  repeated TemplateChoice choices = 5;
+
+  // Interface instances defined in this interface.
+  repeated CoImplements coImplements = 6;
+
+  // View type for this interface
+  Type view = 8;
+
+  repeated TypeConName requires = 9;  // *Available in versions >= 1.dev*
```

Comment:
I think this should only be in `daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf_1.proto`

```suggestion
```